### PR TITLE
fix a couple bugs in archive querying when browser archiving is disabled

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

fixes the following bugs:

* do not do authorized to archive check when browser archiving is disabled in `ArchiveSelector::getArchiveIds()`. that function is only ever called when browser archiving is disabled so the request will never be authorized to archive, and we want to look at the latest invalidated archive if present.
* when selecting archives we want to select all partial archives not just the latest for a plugin. I think those changes were put in to get a test to pass, but since this function didn't have tests didn't notice the problem. This is needed for report specific invalidations.
* in the browser archiving / trigger=archivephp archiving path, the partial archive detection was not pulling partial archives newer than the latest known all plugins archive. fixed and with a test change that shows the error.

CC @tsteur 

### Review

* [ ] Functional review done
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: tests/PHPUnit/Integration/DataAccess/ArchiveSelectorTest.php [hunk 1]
```diff
@@ -30,24 +30,85 @@ protected static function configureFixture($fixture)
...
+                [1],
+                [
+                    ['day', '2020-03-01'],
+                    ['week', '2020-03-02'],
+                ],
+                '',
+                [],
+                [
+                    'done' => [
+                        '2020-03-01,2020-03-01' => [1],
```
Comment (sgiehl) [inline tests/PHPUnit/Integration/DataAccess/ArchiveSelectorTest.php:75]: Maybe I didn't understand the code correctly, but shouldn't that contain the biggest archive id available for that period. Looking at the data above, that seems to be `3` and not `1`

Comment (diosmosis) [reply tests/PHPUnit/Integration/DataAccess/ArchiveSelectorTest.php:75]: The value for `3` is `DONE_ERROR` so we don't use it.

Comment (sgiehl) [reply tests/PHPUnit/Integration/DataAccess/ArchiveSelectorTest.php:75]: ah ok. that makes sense

**Comment by diosmosis** [comment]
@tsteur can you take a look at this PR?

**Comment by tsteur** [comment]
looks good @diosmosis

**Pull request merged**